### PR TITLE
Add `RegionLayouter::constrain_instance` method

### PR DESCRIPTION
**Pull request by therealyingtong**
This allows us to call `region.constrain_instance()` (where previously we had to call `layouter.constrain_instance()`.

### CODE REVIEW

File: src/circuit/floor_planner/single_pass.rs [hunk 1]
```diff
@@ -226,6 +226,20 @@ impl<'r, 'a, F: Field, CS: Assignment<F> + 'a> RegionLayouter<F>
...
+        cell: Cell,
+        instance: Column<Instance>,
+        row: usize,
+    ) -> Result<(), Error> {
+        self.layouter.cs.copy(
+            cell.column,
+            *self.layouter.regions[*cell.region_index] + cell.row_offset,
+            instance.into(),
+            row,
+        )
```
Comment (str4d) [inline src/circuit/floor_planner/single_pass.rs:240]:
```suggestion
        self.layouter.constrain_instance(
            cell,
            instance,
            row,
        )
```
(plus whatever `rustfmt` wants)

File: src/dev.rs [hunk 2]
```diff
@@ -66,7 +66,7 @@ pub enum VerifyFailure {
     /// A permutation did not preserve the original value of a cell.
     Permutation {
         /// The column in which this permutation is not satisfied.
-        column: Column<Any>,
+        column: metadata::Column,
```
Comment (str4d) [inline src/dev.rs:69]: What is the motivation for this change?

**Review by str4d — COMMENTED** [review]
I'm not against this change, but I think it's a sign that the `Region::assign_advice_from_instance` API is misleading. We need it on `Region` in order to be able to assign to a cell within a region (which has no absolute position), but the API should only be used in contexts where we know the region logic is specific to the given circuit (or sufficiently abstracted away), since absolute references to instance column rows will make chips non-portable. We should document these APIs more clearly, and think about how better to handle this abstraction boundary.

File: src/dev.rs (hunk 2 above)
Comment (therealyingtong) [reply src/dev.rs:69]: I want to be able to specify a column in an expected `VerifyFailure` error, without constructing an actual column: https://github.com/zcash/orchard/pull/101/files#diff-a61514f6b9ddb874831490698e7a84231d985315c900accb9c80c302dda52b9aR490

**Comment by therealyingtong** [comment]
I no longer need this after #342.

However, I still need some patch to the MockProver `metadata`: https://github.com/zcash/halo2/pull/343
